.circleci/config.yml

```diff
@@ -66,8 +64,8 @@ jobs:
             cd mattermost-server
             make config-reset
             make check-style BUILD_NUMBER='${CIRCLE_BRANCH}-${CIRCLE_BUILD_NUM}'
-            GOFLAGS=-p=8 make build BUILD_NUMBER='${CIRCLE_BRANCH}-${CIRCLE_BUILD_NUM}'
-            GOFLAGS=-p=8 make package BUILD_NUMBER='${CIRCLE_BRANCH}-${CIRCLE_BUILD_NUM}'
+            GOFLAGS="-p=8 -mod=vendor" make build BUILD_NUMBER='${CIRCLE_BRANCH}-${CIRCLE_BUILD_NUM}'
+            GOFLAGS="-p=8 -mod=vendor" make package BUILD_NUMBER='${CIRCLE_BRANCH}-${CIRCLE_BUILD_NUM}'
```

Comment:
Can we just preserve the original `GOFLAGS` here and append `-p 8`?

Other question: Why are we setting the number of parallel build anyway? Why don't use the default value (`The default is the number of CPUs available.`) @reflog Do you know?

Reply:
no idea :) pinging @cpanato

Reply:
in some tests i did in circleci if not specify that it broke, i will make this test again to check, since w did some fixes.
give me some time to check that

Reply:
@hanzei @reflog here: https://github.com/mattermost/mattermost-server/pull/12799 removed will do the same for enterprise

Reply:
> Can we just preserve the original GOFLAGS here and append -p 8?

Yeah, I was thinking the same thing. I was about to do `GOFLAGS="$GOFLAGS -p=8"`, but then thought this was cleaner.

Anyways, now that https://github.com/mattermost/mattermost-server/pull/12799 is out. Let me remove all this. That should automatically preserve the `-mod=vendor` flag and fix the issue.